### ISSUE TITLE
ImportError on functools.cache on Python 3.8 for role_name rule

### ISSUE THREAD

**Issue by andy-maier** [ansible/ansible-lint#3076]
##### Summary

ansible-lint 6.13.0 fails with ImportError on functools.cache on Python 3.8 for the role_name rule.

ansible-lint 6.12.2 also cannot import functools.cache but it does not fail on Python 3.8 and instead ignores the role_name rule and puts out a warning about that.

The role_name.py file has imported `functools.cache` for some time, but `cache` was added to `functools` only in Python 3.9. Yet, ansible-lint 6.13 requires Python>=3.8.

That means, the change in ansible-lint 6.13.0 was a breaking change for its users on Python 3.8.

For users who need to stay on Python 3.8, the problem can be circumvented by pinning ansible-lint to <6.13 when on Python 3.8. The problem does not exist on Python 3.9 or higher.

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint --version
6.13.0 and 6.13.1
```

- ansible installation method: pip
- ansible-lint installation method: pip

in a virtual env

##### STEPS TO REPRODUCE

On Python 3.8:

```console (paste below)
$ pip install ansible-lint==6.13.0

# have a playbook named playbook.yml that uses a role

$ ansible-lint playbook.yml 
WARNING: ansible-lint is no longer tested under Python 3.8 and will soon require 3.9. Do not report bugs for this version.
Traceback (most recent call last):
  File "/Users/maiera/virtualenvs/vpcauto38/bin/ansible-lint", line 8, in <module>
    sys.exit(_run_cli_entrypoint())
  File "/Users/maiera/virtualenvs/vpcauto38/lib/python3.8/site-packages/ansiblelint/__main__.py", line 359, in _run_cli_entrypoint
    sys.exit(main(sys.argv))
  File "/Users/maiera/virtualenvs/vpcauto38/lib/python3.8/site-packages/ansiblelint/__main__.py", line 235, in main
    rules = RulesCollection(options.rulesdirs, profile_name=options.profile)
  File "/Users/maiera/virtualenvs/vpcauto38/lib/python3.8/site-packages/ansiblelint/rules/__init__.py", line 388, in __init__
    for rule in load_plugins(rulesdirs):
  File "/Users/maiera/virtualenvs/vpcauto38/lib/python3.8/site-packages/ansiblelint/rules/__init__.py", line 339, in load_plugins
    import_module(f"{f.stem}")
  File "/usr/local/opt/python@3.8/Frameworks/Python.framework/Versions/3.8/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 843, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/Users/maiera/virtualenvs/vpcauto38/lib/python3.8/site-packages/ansiblelint/rules/role_name.py", line 26, in <module>
    from functools import cache
ImportError: cannot import name 'cache' from 'functools' (/usr/local/opt/python@3.8/Frameworks/Python.framework/Versions/3.8/lib/python3.8/functools.py)
```

##### Desired Behavior

ansible-lint should not fail when using it on a Python version it claims to be supported (such as ansible-lint 6.13.x claiming support for Python 3.8)

More concretely, I think it should behave as it did with 6.12.2, i.e. to issue a warning about role_name not being used, but otherwise succeeding.

##### Actual Behavior

See above

**Comment by ssbarnea** [comment]
We released 6.14 and at this moment we do not plan to backport a fix for that bug.
